### DISCUSSION TITLE
How do I tell if Nuxt 3 is running in dev mode?

### DISCUSSION THREAD

**Discussion by evrys** [nuxt/nuxt#18779] [Questions]
I looked through the documentation and properties of `useNuxtApp()` but couldn't find a boolean. How do I know if it's dev?

**Comment by danielroe** [comment]
You can use `process.dev` in this case.

**Reply by renansantos** [reply]
Is that in the docs? i couldn't find it there!

**Reply by danielroe** [reply]
Since this answer, we have moved to recommending using `import.meta.dev` instead.

**Reply by PhE** [reply]
Any workaround since `import.meta.dev` [raises an error](nuxt/discussions/24507) ?

**Reply by danielroe** [reply]
Please open an issue with a reproduction. That link leads to a 404.

**Reply by katerlouis** [reply]
@PhE is that the error you were talking about? The component I'm trying to `v-if="!import.meta.dev"` does not use i18n at all and also doesn't complain if I say `v-if="false"`–
![image](https://github.com/user-attachments/assets/017c0c10-8dff-4f02-9bca-bedf128b2e15)

**Comment by evrys** [comment]
Thanks! This works on the client side; however, it seems to be `undefined` on the server, causing a hydration mismatch.

**Comment by evrys** [comment]
Ah, I figured it out: it was undefined because I was destructuring the object like `const { dev } = process`

Writing it as `const dev = process.dev` works correctly on the server, presumably because it's doing something special at compile-time and `process` is not a regular variable
